logger/common.go

```diff
@@ -270,17 +283,24 @@ func (l *Logger) Read(
 
 			// Iteratively scan the unread part in our own buffer and read logs line by line.
 			// Then send it to destination.
-			head := 0
+			head := 0 
 			// This function returns -1 if '\n' in not present in buffer.
 			lenOfLine := bytes.IndexByte(buf[head:bytesInBuffer], newline)
 			for lenOfLine >= 0 {
+				if isPartialMsg {
+					isLastPartial = true
+				} else {
+					msgTimestamp = time.Now().UTC()
+				}
```

Comment:
Why do we need to set the timestamp only when it's not a partial message?

Reply:
Will add a comment. When its partial, all messages that make up that partial should have the same timestamp.